### PR TITLE
tests: stop creating /TEST

### LINKED ISSUE #5856: running test-suite as root creates an empty /TEST directory
### Submission type

  - [X] Bug report
  - [ ] Request for enhancement (RFE)

### systemd version the issue has been seen with

git master

### Used distribution

Debian sid

Running the test suite as root, creates an empty top-level /TEST directory

Steps to reproduce
```
# id
uid=0(root) gid=0(root) groups=0(root)
# meson build
# ninja -C build
# ninja -C build test
# ls -al /TEST/
total 8
drwxr-xr-x  2 root root 4096 Apr 29 13:17 .
drwxr-xr-x 23 root root 4096 Apr 29 13:17 ..
```

### PR DESCRIPTION
**Pull request by evverx**
Closes #5856

### CODE REVIEW

File: test/test-execute/exec-inaccessiblepaths-mount-propagation.service [hunk 1]
```diff
@@ -3,5 +3,5 @@ Description=Test to make sure that InaccessiblePaths= disconnect mount propagati
 
 [Service]
 InaccessiblePaths=-/i-dont-exist
-ExecStart=/bin/sh -x -c 'mkdir -p /TEST; mount -t tmpfs tmpfs /TEST; grep TEST /proc/self/mountinfo && ! grep TEST /proc/$${PPID}/mountinfo && ! grep TEST /proc/1/mountinfo'
+ExecStart=/bin/sh -x -c 'd=$(mktemp -d -p /tmp); trap "umount \'$d\' && rm -r \'$d\'" EXIT; mount -t tmpfs tmpfs "$d"; grep "$d" /proc/self/mountinfo && ! grep "$d" /proc/$${PPID}/mountinfo && ! grep "$d" /proc/1/mountinfo'
```
Comment (evverx) [inline test/test-execute/exec-inaccessiblepaths-mount-propagation.service]: I should probably use `$$()`, `$$`. I will fix.

**Comment by evverx** [comment]
@tixxdz , I'm sorry. I requested a review accidentally, but it would be great if you could review the patch. Thanks.

**Comment by poettering** [comment]
lgtm. but let's see if the CI likes it too

**Comment by poettering** [comment]
the ubuntu CI failure suggests some ubuntu-internal issue. @martinpitt?

**Comment by martinpitt** [comment]
@poettering : yes, some transient cloud failure apparently. I retried the test. (BTW, I sent you info how to do that yourself the other day, did that work?)

**Comment by poettering** [comment]
@martinpitt uh oh, that mail sits in my inbox still, i never actually looked into that. next time!

**Comment by tixxdz** [comment]
@evverx lgtm! oops sorry for pushing this code, I thought that tests where running inside namespaces or something then cleaned... thank you fox fixing it.

**Comment by keszybz** [comment]
`× xenial-i386` — haveged cannot be installed.

**Pull request merged**